sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -0,0 +1,202 @@
+
+      // user
+      @Nullable User user = scope.getUser();
+      if (user == null) {
+        user = new User();
+        scope.setUser(user);
+      }
+      if (user.getId() == null) {
+        try {
+          user.setId(Installation.id(context));
+        } catch (RuntimeException e) {
+          logger.log(SentryLevel.ERROR, "Could not retrieve installation ID", e);
+        }
+      }
+
+      // app context
+      @Nullable App app = scope.getContexts().getApp();
+      if (app == null) {
+        app = new App();
+        app.setAppName(ContextUtils.getApplicationName(context, options.getLogger()));
+        app.setAppStartTime(DateUtils.toUtilDate(AppStartState.getInstance().getAppStartTime()));
+
+        final @NotNull BuildInfoProvider buildInfoProvider =
+            new BuildInfoProvider(options.getLogger());
+        final @Nullable PackageInfo packageInfo =
+            ContextUtils.getPackageInfo(
+                context, PackageManager.GET_PERMISSIONS, options.getLogger(), buildInfoProvider);
+        if (packageInfo != null) {
+          ContextUtils.setAppPackageInfo(packageInfo, buildInfoProvider, app);
+        }
+        scope.getContexts().setApp(app);
+      }
+
+      writer.name("user").value(logger, scope.getUser());
+      writer.name("contexts").value(logger, scope.getContexts());
+      writer.name("tags").value(logger, scope.getTags());
+      writer.name("extras").value(logger, scope.getExtras());
+      writer.name("fingerprint").value(logger, scope.getFingerprint());
+      writer.name("level").value(logger, scope.getLevel());
+      writer.name("breadcrumbs").value(logger, scope.getBreadcrumbs());
+    } catch (Exception e) {
+      options.getLogger().log(SentryLevel.ERROR, "Could not serialize scope.", e);
+      return new HashMap<>();
+    }
+
+    return data;
+  }
+
+  /**
+   * Captures the provided envelope. Compared to {@link IHub#captureEvent(SentryEvent)} this method
+   * - will not enrich events with additional data (e.g. scope) - will not execute beforeSend: it's
+   * up to the caller to take care of this - will not perform any sampling: it's up to the caller to
+   * take care of this - will enrich the envelope with a Session updates is applicable
+   *
+   * @param envelopeData the serialized envelope data
+   * @return The Id (SentryId object) of the event
+   * @throws Exception In case the provided envelope could not be parsed / is invalid
+   */
+  public static SentryId captureEnvelope(final @NotNull byte[] envelopeData) throws Exception {
+    final @NotNull IHub hub = HubAdapter.getInstance();
+    final @NotNull SentryOptions options = hub.getOptions();
+
+    final @NotNull ISerializer serializer = options.getSerializer();
+    final @Nullable SentryEnvelope envelope =
+        options.getEnvelopeReader().read(new ByteArrayInputStream(envelopeData));
+    if (envelope == null) {
+      throw new IllegalArgumentException("Envelope could not be read");
```

Comment:
M: I'd rather prefer that the Java SDK swallow the exception and log it out, and return an empty ID, otherwise handling this use case has to be duplicated across different hybrid SDKs.